### DISCUSSION TITLE
Return banner adsize on banner load

### DISCUSSION THREAD

**Discussion by merbin2012** [EMI-INDO/emi-indo-cordova-plugin-admob#32] [Ideas]
The banner ad overflow issue still occurs on some devices (e.g., Redmi Note 5 Pro). This problem might be due to a delay caused by lower-end mobile hardware.

Can you please add the following feature. Return the banner ad size to the `on.banner.load` event. like below

```
val bannerHeight=adSize.height;

val bannerLoadEventData = String.format(
  "{\"height\": %d}",
  bannerHeight
 )

//cWebView!!.loadUrl("javascript:cordova.fireDocumentEvent('on.banner.load');")

cWebView!!.loadUrl("javascript:cordova.fireDocumentEvent('on.banner.load', $bannerLoadEventData);")
```

So it is easy to handle the banner overflow problem from javascript code.
The sample code is 

call the method in banner.load
```
document.addEventListener('on.banner.load', (arg) => {

		adjustWebViewForBanner(arg.height)
})
```

This will automatically adjust the height
```
async function adjustWebViewForBanner(adHeight) {
	const bannerHeightPx = adHeight;
	const adjustedHeight = window.innerHeight - bannerHeightPx;

	// Adjust the WebView height
	document.body.style.height = `${adjustedHeight}px`;
	document.body.style.overflow = "hidden";
 
  }
```

I have encountered several policy violation issues related to the absence of a black border above the banner ad, categorized as "Encouraging Accidental Clicks: Layout." This issue was not limited to a single app but occurred across multiple apps. I resolved it by adding a 3px black line above the banner ad. If you need the code for this fix, you can refer to the example below.
![Encouraging Accidental clicks - Layout](https://github.com/user-attachments/assets/6e0aaf00-2e90-4013-a48d-9c98bca1f36c)

```
async function adjustWebViewForBanner(adHeight) {

	const bannerHeightPx = adHeight; 
	const lineHeightPx = 3; // Height of the black line
	
	const adjustedHeight = window.innerHeight - bannerHeightPx - lineHeightPx;

	// Adjust the WebView height
	document.body.style.height = `${adjustedHeight}px`;
	document.body.style.overflow = "hidden"; // Prevent content from overflowing
  
	// Add the black line below the banner
	let blackLine = document.getElementById("bannerBottomWhiteSpace");
	if (!blackLine) {
	  blackLine = document.createElement("div");
	  blackLine.id = "bannerBottomWhiteSpace";
	  blackLine.style.width = "100%";
	  blackLine.style.height = `${lineHeightPx}px`;
	  blackLine.style.background = "black";
	  blackLine.style.position = "fixed";
	  blackLine.style.bottom = `${bannerHeightPx}px`; // Position above the banner
	  blackLine.style.zIndex = "99999"; // Ensure it's above the banner
	  document.body.appendChild(blackLine);
	}

  }
```

**Comment by EMI-INDO** [comment]
@merbin2012 

I just debugged it, it doesn't seem to return the actual ad height.

`size: "large_banner" output ad height: 64`
`size: "banner" output ad height: 64`

I didn't test it on a real device

<img width="731" alt="h2" src="https://github.com/user-attachments/assets/85c359e1-f920-4b4a-882a-f0a2dc239de6">

<img width="783" alt="h1" src="https://github.com/user-attachments/assets/1fca3635-dc6f-4e03-937b-74c4c06278d3">

**Reply by merbin2012** [reply]
How did you confirm the size is wrong?

I have used the following banner type
```
const bannerTypeAndroid="anchored_adaptive";
const bannerTypeiOS="responsive_adaptive"
```
And I have tested in Redmi 11T pro, Samsung M34 & iOS (iPad mini is 8.3). It is working.

I have tested and confirmed the banner size by using the following method.

![Banner Height](https://github.com/user-attachments/assets/d6f6fc9d-ccc6-44ae-b23a-ed3e15cd1f3a)

![Banner Height 1](https://github.com/user-attachments/assets/1c469931-d6d3-4158-8c5a-9534dbf91e1e)


I will check with other banner type and let you know.

**Comment by merbin2012** [comment]
I checked it is return the correct value, if it is only "anchored_adaptive". For other banner types, need to add some logic like 
`"large_banner" -> AdSize.LARGE_BANNER`  
Can you please add this too?

In framework7 very easy to manage the overlay issue, by adding the following line
`document.documentElement.style.setProperty('--f7-safe-area-bottom', bannerAdHeightPx+"px");`
